docs/adr/0002-use-local-datastore.md

```diff
@@ -0,0 +1,58 @@
+# 2: Use local datastore for publisher workflow
+
+Date: 2018-07-20
+
+## Context
+We're building a new publishing architecture and MVP workflow in private beta. We have taken the decision to build this in a new application and therefore we need to decide how we use the publishing-api.
+
+One of the goals of the migration effort from 2016 was to for all publishing applications to be clients of the publishing-api and this API would means that these applications would not need local datastores. This is written up in the [GOV.UK Platform Architecture Goals 2017-18+][goals]
+
+[goals]: https://docs.google.com/document/d/1Oft4akc6dZfhhOjosNPbFpcLUOUjz7YG7QPcVZi8hww/edit#heading=h.5uytjxbfoe58
+
+Since writing these goals we have built the specialist-publisher application using this pattern. The difficulties of building Rails application this way have led us to reconsider this approach.
+
+We are about to build another publishing-api client and we should now use the learnings we have to ensure that not using a local datastore is the best decision.
+
+### Option 1: No local datastore
+In this option we continue with the platform architecture goals.
+
+#### Pros
++ Single source of truth - all data is in the publishing api
++ Multiple apps can edit the same content
++ Single implementation of scheduling and publishing logic
++ Single abstraction layer
+
+#### Cons
+- Effectively building an Active Resource layer
+- Harder to resolve performance problems
+- Current workflow is opinionated and may not match needs of content
```

Comment:
How does this related to where the data is stored? Even if you were to keep a copy of the content locally, you'd still be using the Publishing API right, and thus still be involved and bound somewhat by it's supported workflow.

Reply:
If all of the data is stored in the publishing API then it's bound by it's workflow. Where's if you're building you're own data locally then it's possible to deviate from the workflow to better meet the needs of the content. Admittedly this workflow still needs to fit into the publishing-apis, but with the current functionality we can build support for 2i or fact check in the local app.

Reply:
I guess I don't see the Publishing API being limiting. Particularly in the case of things like 2i or fact check, I can imagine ways to implement that without needing to keep a copy of the content locally in the publishing application.

Reply:
I think you're correct that we could build a review process into the Publishing API. But the additional complexity added to the API and additional time needed for development would out weigh the benefits. In this quarters mission we need to iterate and build the new workflow quickly, trying to build this in the Publishing API is probably not the best use of our time.

Reply:
Personally, I actually dislike the idea of adding any functionality around review processes in to the Publishing API.

In my last comment, what I was probably unclear about is that I can imaging keeping the content in the Publishing API, and keeping the details of fact check/2i review in the publishing app, so building _on_ the Publishing API, instead of building _in_ the Publishing API.